client/src/AddFriend.js

```diff
@@ -97,14 +103,40 @@ class AddFriend extends React.Component {
             onChange={this.handleChange}
             placeholder='Friends email'
           />
-          <input type='submit' value='Add Friend' />
+          {!this.state.isloading && submitButton()}
+          {this.state.isloading && loadButton()}
         </form>
-
+            
         <h5>Total Count</h5>
         <h5>0</h5>
       </div>
     )
   }
 }
 
+function loadButton() {
+  return (
+    <>
+  <Button variant="secondary" disabled>
+    <Spinner
+      as="span"
+      animation="border"
+      size="sm"
+      role="status"
+      aria-hidden="true"
+    />
+    Loading...
+  </Button>
+</>
+  
```

Comment:
Another way to do this is to have one button instead of two, and rather than using the loadingState to toggle the entire button, you can use it to just toggle the spinner + text + whether it's disabled. It's mainly a style choice though.